src/map/pc.c

```diff
@@ -4052,7 +4052,21 @@ static int pc_bonus3(struct map_session_data *sd, int type, int type2, int type3
 			}
 
 			break;
+		case SP_STATE_NO_RECOVER_RACE:
+		{
+			uint32 race_mask = map->race_id2mask(type2);
```

Comment:
look like wrong validation for type2
if ``race > RC_FORMLESS``, map->race_id2mask will return ``1 << race``. for example for ``RC_NONBOSS`` it will be ``1 << 12`` and this equal ``4096``

but size of no_recover_state_race up to RC_MAX